cmd/dra-example-controller/driver.go

```diff
@@ -93,10 +93,24 @@ func (d driver) GetClaimParameters(ctx context.Context, claim *resourcev1.Resour
 }
 
 func (d driver) Allocate(ctx context.Context, claim *resourcev1.ResourceClaim, claimParameters interface{}, class *resourcev1.ResourceClass, classParameters interface{}, selectedNode string) (*resourcev1.AllocationResult, error) {
+
 	if selectedNode == "" {
-		return nil, fmt.Errorf("TODO: immediate allocations not yet supported")
+		allocationResult, err := d.allocateImmediateClaim(claim, claimParameters, class, classParameters)
+		if err != nil {
+			return nil, fmt.Errorf("error performing immediate allocation: %v", err)
+		}
+		return allocationResult, nil
 	}
 
+	allocationResult, err := d.allocateDelayedClaim(claim, claimParameters, class, classParameters, selectedNode)
+	if err != nil {
+		return nil, fmt.Errorf("error performing delayed allocation: %v", err)
```

Comment:
Can we just trust the code of local functions and simply return errors as they are? This also applies to the `allocateImmediateClaim ` call above.

Reply:
It's not about "trusting" it. It's about consistency and being able to trace the errors through the function calls when you see them appear in the logs.

Reply:
I specifically requested @byako to add these wrappers